### PR TITLE
unix: Add NetBSD bits in core.c

### PR DESCRIPTION
**Pull request by krytarowski**
NetBSD supports:
 - O_CLOEXEC
 - accept4(2) like API with paccept(2)
   NetBSD 8.0 will get compat accept4(2)
 - SOCK_NONBLOCK
 - SOCK_CLOEXEC
 - dup3(2)

### CODE REVIEW

File: src/unix/core.c [hunk 1]
```diff
@@ -462,7 +472,7 @@ int uv__accept(int sockfd) {
   assert(sockfd >= 0);
 
   while (1) {
-#if defined(__linux__) || (defined(__FreeBSD__) && __FreeBSD__ >= 10)
+#if defined(__linux__) || (defined(__FreeBSD__) && __FreeBSD__ >= 10) || defined(__NetBSD__)
```
Comment (cjihrig) [inline src/unix/core.c]: This line is over 80 characters now.

File: src/unix/core.c [hunk 2]
```diff
@@ -73,6 +73,16 @@
 # endif
 #endif
 
+#if defined(__NetBSD__)
+# include <sys/sysctl.h>
+# include <sys/filio.h>
+# include <sys/wait.h>
+# define UV__O_CLOEXEC O_CLOEXEC
+# define uv__accept4(a,b,c,d) paccept((a),(b),(c),NULL,(d))
```
Comment (cjihrig) [inline src/unix/core.c]: Spaces after the commas please.

File: src/unix/core.c [hunk 3]
```diff
@@ -73,6 +73,16 @@
 # endif
 #endif
 
+#if defined(__NetBSD__)
```
Comment (cjihrig) [inline src/unix/core.c]: It looks like this block could be combined with the block before it.

File: src/unix/core.c (hunk 1 above)
Comment (txdv) [reply src/unix/core.c]: break it up with \ at the end and the new statement in a new line?

**Comment by krytarowski** [comment]
Should be done.

**Comment by santigimeno** [comment]
CI: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/482/

**Comment by cjihrig** [comment]
Landed in https://github.com/libuv/libuv/commit/c0504885ea4a7f763a5c33b5dda8dfd9d42c5be6. Thanks!

**Pull request closed without merging**